website/docs/transforms/rename.md

```diff
@@ -17,37 +17,137 @@ Add the following configuration to your Mesh config file:
+            field: api(.*)
+          to:
+            type: Query
+            field: $1
+          useRegExpForFields: true
 
-      - from: 
-          type: Query
-          field: api(.*)
-        to: 
-          type: Query
-          field: $1
-        useRegExpForFields: true
+```
+
+## Two different modes
+Although apparently both `bare` and `wrap` modes do achieve the same result, when it comes to renaming; they are very different.  
+Let's take a look at how they operate.
+
+### Wrap
+Wrap applies transformations by adding a wrapping layer to the original GraphQL schema. The handler generates a GraphQL
+ schema and passes it to the rename transform. `wrap` receives this schema and rather than updating it, it will apply a layer on top of it, which scope is to provide the values exposed by the names of the original schema to the new names that you define as `renames`.  
+This approach is safe as we have used it extensively in `graphql-tools`. However, it might have some runtime implications, since when receiving requests and preparing responses, the wrapping layer needs to apply the `renames` on the fly for every GraphQL operation.
+
+> NOTE: `wrap` is the only approach that works when you want to transform at all-sources (root) level, unless you're using [merger-bare](https://graphql-mesh.com/docs/api/modules/merger-bare/). If you want to remove the possible runtime implications, consider either moving your `renames` at the data source level or opt into `merger-bare`; in order to take advantage of `bare` mode.
+
+Example:
+```yaml
+sources:
+  - name: Countries
+    handler:
+      soap:
+        wsdl: http://webservices.../wso?WSDL
+  - name: Users
+    handler:
+      openapi:
+        source: https://api.../swagger.yaml
+    transforms:
+      rename:
+          mode: wrap # you can use wrap here, but consider using "bare" to remove runtime cost
+          renames:
+            - from:
+                type: User
+                field: lastName
+              to:
+                type: User
+                field: surname
+transforms:
+  rename:
+      mode: wrap # bare won't work here at all-sources (root) level, because you're not using merger-bare
+      renames:
+        - from:
+            type: Country
+            field: ISO-3166_Code
+          to:
+            type: Country
+            field: code
+```
 
+> ProTip: When you want to use `wrap`, you can omit the "mode" property since this is already applied by default.  
+
+### Bare
+Bare is a recent addition and works by applying transformations to the original schema. The handler generates a GraphQL schema and passes it to the rename transform. `bare` will rename types and fields as defined in the schema generated by your handler and will return an updated version of the same schema with the transformations applied.  
+The advantage of `bare` is that it commits the schema as part of the initialisation process only; then it won't have any runtime cost.  
```

Comment:
If you don't want to talk about "cost", consider changing this to "implications" as well; for consistency.